src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotContainer;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.XboxController;
+ 
+public class Limelight extends SubsystemBase {
+  private NetworkTable table;
+    
+    private double tx;
+    private double ty;
+    private double apriltagid;
+   
+    
+  /** Creates a new Limelight. */
+  public Limelight() {
+    table = NetworkTableInstance.getDefault().getTable("limelight");
+    disableLimelight();
+    
+    
+  }
+  
+public void enableLimelight() {
+  table.getEntry("ledMode").setNumber(3);
+}
+public void disableLimelight() {
+  table.getEntry("ledMode").setNumber(1);
+}
+
+  @Override
+  public void periodic() {
+    tx = table.getEntry("tx").getDouble(0.0);
+    ty = table.getEntry("ty").getDouble(0.0);
+    
+
+   
+
+    apriltagid = table.getEntry("tid").getDouble(0.0);
+    
+    SmartDashboard.putNumber("x offset", tx);
+    SmartDashboard.putNumber("y offset", ty);
+
+    SmartDashboard.putNumber("TId", apriltagid);
+
+    limelightCenter();
```

Comment:
/*
     * Instead of putting this action in a method, it should go into a separate command. Why?
     * 1. This action requires more than one subsystem. We shouldn't be accessing other subsystems from within this subsystem
     * 2. This action needs to be triggered by a particular action, not constantly running by default. We can configure
     * a command to be bound a trigger
     */